.github/workflows/ci.yml

```diff
@@ -1,18 +1,31 @@
-name: CI
+name: Test
 
-on: [push]
+on: [push, pull_request, workflow_call]
 
 jobs:
-  test:
+  qa:
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: pre-commit/action@v3.0.0
+
+  test:
+    needs: qa
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest]
+        python-version: [ "3.9", "3.10", "3.11" ]
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python
-      uses: actions/setup-python@v2
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v4
       with:
-        python-version: 3.10.9
+        python-version: ${{ matrix.python-version }}
+  
     - name: Install and build
-      run: pip install -r requirements-dev.txt
+      run: python -m pip install -r requirements-dev.txt
```

Comment:
Just curious what's the difference between pip install and python -m pip install?

Reply:
Have a look at https://stackoverflow.com/questions/25749621/whats-the-difference-between-pip-install-and-python-m-pip-install

We might not need it in this case, but it is becoming more common practice to ensure consistency.